ocs_ci/utility/utils.py

```diff
@@ -2282,17 +2287,47 @@ def ceph_health_check(namespace=None, tries=20, delay=30):
     """
     if config.ENV_DATA["platform"].lower() == constants.IBM_POWER_PLATFORM:
         delay = 60
-    return retry(
-        (
-            CephHealthException,
-            CommandFailed,
-            subprocess.TimeoutExpired,
-            NoRunningCephToolBoxException,
-        ),
-        tries=tries,
-        delay=delay,
-        backoff=1,
-    )(ceph_health_check_base)(namespace)
+
+    if (
+        cluster_type == constants.CLUSTER_TYPE_EXTERNAL
+        or cluster_type == constants.CLUSTER_TYPE_INTERNAL
+    ):
+        # checking if external mode is standalone or multi-storagecluster to determine namespace
+        if namespace is None:
+            if cluster_type == constants.CLUSTER_TYPE_EXTERNAL:
+                if config.DEPLOYMENT.get("multi_storagecluster"):
+                    namespace = constants.OPENSHIFT_STORAGE_EXTENDED_NAMESPACE
+                else:
+                    namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+            else:
+                namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
```

Comment:
```suggestion
        if namespace is None:
            if cluster_type == constants.CLUSTER_TYPE_EXTERNAL:
                if config.DEPLOYMENT.get("multi_storagecluster"):
                    namespace = constants.OPENSHIFT_STORAGE_EXTENDED_NAMESPACE
```

The namespace is correctly taken from configuration in ceph_healht_check_base here:

https://github.com/shyRozen/ocs-ci/blob/9ca9c4e26e8fd361962d410d2e297f3bfeddb609/ocs_ci/utility/utils.py#L2349

For regular deployment, namespace should always be taken from config, as in fusion cluster deployment, there is different namespace for ODF used, and we are relaying on config value, and should not on constants.
@fbalak  tried to fix that on most of the places as part of his [PR](https://github.com/red-hat-storage/ocs-ci/pull/7501/files) some time back, this will break the logic.